### PR TITLE
[monitor query] move metrics dataplane to new azmetrics module

### PR DESCRIPTION
**Pull request by gracewilcox**
Moving the metrics data plane functionality to a new module. Part of a wider Monitor restructuring.

Part of #22377

### CODE REVIEW

**Comment by gracewilcox** [comment]
go - azmetrics - ci

File: sdk/monitor/query/azmetrics/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Azure Monitor Metrics client module for Go
+
+Source code | Package (pkg.go.dev) | [REST API documentation][monitor_rest_docs] | [Product documentation][monitor_docs] | Samples
+
+## Getting started
+
+### Prerequisites
+
+* Go, version 1.18 or higher - [Install Go](https://go.dev/doc/install)
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/README.md:9]:
```suggestion
* Go version 1.18 or higher - [Install Go](https://go.dev/doc/install)
```

File: sdk/monitor/query/azmetrics/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+### Install the packages
+
+Install the `azmetrics` and `azidentity` modules with `go get`:
+
+```bash
+go get github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azmetrics
+go get github.com/Azure/azure-sdk-for-go/sdk/azidentity
+```
+
+The [azidentity][azure_identity] module is used for Azure Active Directory authentication during client construction.
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/README.md:22]:
```suggestion
The [azidentity][azure_identity] module is used for client authentication.
```

File: sdk/monitor/query/azmetrics/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+This project has adopted the [Microsoft Open Source Code of Conduct][coc]. For more information, see
+the [Code of Conduct FAQ][coc_faq] or contact [opencode@microsoft.com][coc_contact] with any additional questions or
+comments.
+
+<!-- LINKS -->
+[azure_identity]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity
+[azure_sub]: https://azure.microsoft.com/free/
+[azure_monitor_overview]: https://learn.microsoft.com/azure/azure-monitor/overview
+[context]: https://pkg.go.dev/context
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/README.md:67]:
```suggestion
```
(unused)

File: sdk/monitor/query/azmetrics/custom_client.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+// A credential with authorization at the subscription level is required when using this client.
+//
+// endpoint - The regional endpoint to use, for example https://eastus.metrics.monitor.azure.com.
+// The region should match the region of the requested resources. For global resources, the region should be 'global'.
+func NewClient(endpoint string, credential azcore.TokenCredential, options *ClientOptions) (*Client, error) {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+
+	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{"https://metrics.monitor.azure.com" + "/.default"}, nil)
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/custom_client.go:33]:
```suggestion
	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{"https://metrics.monitor.azure.com/.default"}, nil)
```

File: sdk/monitor/query/azmetrics/utils_test.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azmetrics"
+	"github.com/stretchr/testify/require"
+)
+
+const recordingDirectory = "sdk/monitor/query/azmetrics/testdata"
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/utils_test.go:26]:
```go
const (
     ...
)
```

File: sdk/monitor/query/azmetrics/utils_test.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+func startTest(t *testing.T) *azmetrics.Client {
+	startRecording(t)
+	transport, err := recording.NewRecordingHTTPClient(t, nil)
+	require.NoError(t, err)
+	opts := &azmetrics.ClientOptions{ClientOptions: azcore.ClientOptions{Transport: transport}}
+	endpoint := "https://" + region + ".metrics.monitor.azure.com"
+	client, err := azmetrics.NewClient(endpoint, credential, opts)
+	if err != nil {
+		panic(err)
+	}
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/utils_test.go:95]:
```suggestion
	require.NoError(t, err)
```

File: sdk/monitor/query/azmetrics/utils_test.go [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+	return client
+}
+
+func getEnvVar(envVar string, fakeValue string) string {
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/utils_test.go:99]: Would [recording.GetEnvVar()](https://github.com/Azure/azure-sdk-for-go/blob/b08f175d31a9a7d48edbb8f76ce1a1721f2e4232/sdk/internal/recording/recording.go#L131) work?

File: sdk/monitor/query/azmetrics/utils_test.go (hunk 6 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/utils_test.go]: Fixed in 74e4dae

File: sdk/monitor/query/azmetrics/utils_test.go (hunk 5 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/utils_test.go]: Fixed in 74e4dae

File: sdk/monitor/query/azmetrics/custom_client.go (hunk 4 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/custom_client.go]: Fixed in 74e4dae

File: sdk/monitor/query/azmetrics/README.md (hunk 3 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/README.md]: Fixed in 74e4dae

File: sdk/monitor/query/azmetrics/README.md (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/README.md]: Fixed in 74e4dae

File: sdk/monitor/query/azmetrics/README.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/README.md]: Fixed in 74e4dae

File: sdk/monitor/query/azmetrics/utils_test.go (hunk 7 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/utils_test.go:99]: Would need to look into. Maybe in a separate PR?

File: sdk/monitor/query/azmetrics/README.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+<!-- LINKS -->
+[azure_identity]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity
+[azure_sub]: https://azure.microsoft.com/free/
+[azure_monitor_overview]: https://learn.microsoft.com/azure/azure-monitor/overview
+[cloud_documentation]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud
+[default_cred_ref]: https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/azidentity#defaultazurecredential
+[go_samples]: (https://github.com/Azure-Samples/azure-sdk-for-go-samples)
+[metrics_overview]: https://learn.microsoft.com/azure/azure-monitor/essentials/data-platform-metrics
+[metric_namespaces]: https://learn.microsoft.com/azure/azure-monitor/reference/supported-metrics/metrics-index#metrics-by-resource-provider
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/README.md:71]:
```suggestion
```

File: sdk/monitor/query/azmetrics/README.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+do this once across all repos using our CLA.
+
+This project has adopted the [Microsoft Open Source Code of Conduct][coc]. For more information, see
+the [Code of Conduct FAQ][coc_faq] or contact [opencode@microsoft.com][coc_contact] with any additional questions or
+comments.
+
+<!-- LINKS -->
+[azure_identity]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity
+[azure_sub]: https://azure.microsoft.com/free/
+[azure_monitor_overview]: https://learn.microsoft.com/azure/azure-monitor/overview
```
Comment (chlowell) [inline sdk/monitor/query/azmetrics/README.md:66]:
```suggestion
```

**Review by chlowell — APPROVED** [review]

File: sdk/monitor/query/azmetrics/README.md (hunk 9 above)
Comment (gracewilcox) [reply sdk/monitor/query/azmetrics/README.md]: fixed in 44be96d

**Pull request merged**